tests/batch/test_AnalyserAndReporterWaqua.py

```diff
@@ -0,0 +1,152 @@
+from unittest.mock import patch
+import pytest
+import numpy
+from dfastmi.batch.AnalyserAndReporterWaqua import ReporterWaqua, OutputDataWaqua
+from dfastmi.batch.AnalyserAndReporterWaqua import AnalyserWaqua
+        
+class Test_ReporterWaqua():
+    def given_output_data_and_mocked_write_files_when_write_report_then_expect_3_calls_for_writing_report(self):
+        reporter = ReporterWaqua("filepath")
+        firstm = 0
+        firstn = 0
+        data_zgem = numpy.array([1, 2, 3, 4, 5])
+        data_zmax = numpy.array([1, 2, 3, 4, 5])
+        data_zmin = numpy.array([1, 2, 3, 4, 5])
+        output_data = OutputDataWaqua(firstm, firstn, data_zgem, data_zmax, data_zmin)
+        with patch('dfastmi.batch.AnalyserAndReporterWaqua.ApplicationSettingsHelper.get_filename') as mocked_get_file_name,\
+        patch('dfastmi.batch.AnalyserAndReporterWaqua.DataTextFileOperations.write_simona_box') as mocked_write_simona_box:
+            
+            mocked_get_file_name.return_value = "filename.file"
+            mocked_get_file_name.side_effect = "file.file"
+    
+            reporter.write_report(output_data)
+            
+            assert mocked_write_simona_box.call_count == 3
+            assert mocked_get_file_name.call_count == 3
+
+class Test_AnalyserWaqua():
+    
+    @pytest.mark.parametrize("display, reduced_output, old_zmin_zmax", [
+        (False, False, False),
+        (True, False, False),
+        (True, True, False),
+        (True, True, True),
+        (False, True, False),
+        (False, True, True),
+        (False, False, True),
+        (True, False, True),
+    ])
+    def given_data_and_mocked_classes_with_missing_file_location_when_analyse_then_return_success_and_empty_output_data(self, display : bool, reduced_output : bool, old_zmin_zmax : bool):
+        tstag = 0.0
```

Comment:
Unclear because I forgot.. but wasn't it agreed between us to add a text here splitting the name in GWT:
```suggestion
     """"
     given : data and mocked classes with missing filelocation
     when :  analyse 
     then  : return success and empty output_data
     """
        tstag = 0.0
```

Reply:
done